### ISSUE TITLE
Using an enum class as a template parameter will silently break reflection

### ISSUE THREAD

**Issue by cbsirb** [stephenberry/glaze#631]
Given this:
```
#include <iostream>
#include <glaze/glaze.hpp>

enum class BType {
    Type1,
    Type2,
};

template<BType T, typename DataType>
class B {
public:
    BType type { T };
    DataType data;

    B(DataType data_) : data(data_) {}

    struct glaze {
        static constexpr auto value = glz::object(&B::type, &B::data);
    };
};

int main() {
    B<ActualType::Type1, int> b(3);
    std::cout << glz::write_json(b) << "\n";

    return 0;
}
```
It will print out `{"Type1, int>::type":0,"Type1, int>::data":3}`. Basically it will only take the type until the first `::` pair.

I don't know if this should be supported or not (it's a weird and very specific case), but it should be mentioned in the documentation at least.

**Comment by stephenberry** [comment]
Thanks for discovering this! Will look into it.

**Comment by stephenberry** [comment]
This was an issue with the member object pointer reflection logic. It has been fixed with #672
